packages/components/src/disabled/test/index.tsx

```diff
@@ -134,6 +135,43 @@ describe( 'Disabled', () => {
 		);
 	} );
 
+	it( 'should preserve input values when toggling the isDisabled prop', async () => {
+		const user = userEvent.setup( {
+			advanceTimers: jest.advanceTimersByTime,
+		} );
+
+		const MaybeDisable = ( { isDisabled = true } ) => (
+			<Disabled isDisabled={ isDisabled }>
+				<Form />
+			</Disabled>
+		);
+
+		const getInput = () => screen.getByRole( 'textbox' );
+		const getContentEditable = () => screen.getByTitle( 'edit my content' );
+
+		const { rerender } = render( <MaybeDisable isDisabled={ false } /> );
+
+		await user.type( getInput(), 'This is input.' );
+		expect( getInput() ).toHaveValue( 'This is input.' );
+
+		await user.type( getContentEditable(), 'This is contentEditable.' );
+		expect( getContentEditable() ).toHaveTextContent(
+			'This is contentEditable.'
+		);
+
+		rerender( <MaybeDisable isDisabled={ true } /> );
+		expect( getInput() ).toHaveValue( 'This is input.' );
+		expect( getContentEditable() ).toHaveTextContent(
+			'This is contentEditable.'
+		);
```

Comment:
If an instance previously obtained by `screen.getByRole( 'textbox' )` is used, `rerender` is not reflected, so a new instance is obtained.